lib/debloater/cli.rb

```diff
@@ -0,0 +1,92 @@
+require 'debloater/connection'
+require 'debloater/engine'
+require 'io/console'
+
+module Debloater
+  class CLI
+    DEFAULTS = {
+      connection: {
+        host:     'localhost',
+        port:     5432,
+        user:     'postgres',
+        password: nil,
+        dbname:   nil,
+      },
+      engine: {
+        confirm:    true,
+        min_mb:     50,
+        max_density: 0.75,
+      },
+      prompt_password: true,
+    }
+
+    def initialize(argv)
+      @options = DEFAULTS.dup
+      _parse(argv.dup, @options)
+    end
+    
+    def run
+      conn = Connection.new(@options[:connection])
+      Engine.new(conn, @options[:engine]).run
+    end
+
+    private
+
+    def _parse(argv, options)
```

Comment:
This method call feels a bit weird to me; had to do a triple-take before I worked out that `options` was essentially being used as a mutable ref parameter. I'd be inclined to do `options = DEFAULTS.dup` as the first line in this method and then return it as the result to make things a little more readable.